src/Storages/System/StorageSystemDictionaries.cpp

```diff
@@ -70,96 +135,84 @@ void StorageSystemDictionaries::fillData(MutableColumns & res_columns, ContextPt
         }
 
-        res_columns[i++]->insert(static_cast<UInt64>(std::chrono::system_clock::to_time_t(load_result.loading_start_time)));
-        res_columns[i++]->insert(static_cast<UInt64>(std::chrono::system_clock::to_time_t(load_result.last_successful_update_time)));
-        res_columns[i++]->insert(std::chrono::duration_cast<std::chrono::duration<float>>(load_result.loading_duration).count());
-
-        if (last_exception)
-            res_columns[i++]->insert(getExceptionMessage(last_exception, false));
-        else
-            res_columns[i++]->insertDefault();
-
-        if (dict_ptr)
+        if (dict_structure)
         {
-            res_columns[i++]->insert(dict_ptr->getDictionaryComment());
-        }
-        else
-        {
-            if (load_result.config && load_result.config->config->has("dictionary.comment"))
-                res_columns[i++]->insert(load_result.config->config->getString("dictionary.comment"));
+            res_columns[SystemDictionariesColumn::KeyNames]->insert(
+                collections::map<Array>(dict_structure->getKeysNames(), [](auto & name) { return name; }));
+
+            if (dict_structure->id)
+                res_columns[SystemDictionariesColumn::KeyTypes]->insert(Array({"UInt64"}));
             else
-                res_columns[i++]->insertDefault();
+                res_columns[SystemDictionariesColumn::KeyTypes]->insert(
+                    collections::map<Array>(*dict_structure->key, [](auto & attr) { return attr.type->getName(); }));
+
+            res_columns[SystemDictionariesColumn::AttributeNames]->insert(
+                collections::map<Array>(dict_structure->attributes, [](auto & attr) { return attr.name; }));
+            res_columns[SystemDictionariesColumn::AttributeTypes]->insert(
+                collections::map<Array>(dict_structure->attributes, [](auto & attr) { return attr.type->getName(); }));
+            res_columns[SystemDictionariesColumn::Key]->insert(dict_structure->getKeyDescription());
         }
 
-        /// Start fill virtual columns
+        res_columns[SystemDictionariesColumn::LoadingStartTime]->insert(
+            static_cast<UInt64>(std::chrono::system_clock::to_time_t(load_result.loading_start_time)));
+        res_columns[SystemDictionariesColumn::LastSuccessfulUpdateTime]->insert(
+            static_cast<UInt64>(std::chrono::system_clock::to_time_t(load_result.last_successful_update_time)));
+        res_columns[SystemDictionariesColumn::LoadingDuration]->insert(
+            std::chrono::duration_cast<std::chrono::duration<float>>(load_result.loading_duration).count());
 
-        res_columns[i++]->insert(dictionary_structure.getKeyDescription());
+        if (last_exception)
+            res_columns[SystemDictionariesColumn::LastException]->insert(getExceptionMessage(last_exception, false));
+
+        index++;
+        /// Set to default unfilled columns
```

Comment:
Better to rename something like `Add default value to unfilled columns`.

Reply:
Changed the code. No longer relevant